File: sol/solution/0201-0300/0202/Solution.go
func isHappy(n int) bool {
    getNext := func(num int) int {
        sum := 0
        for num > 0 {
            d := num % 10
            sum += d * d
            num /= 10
        }
        return sum
    }
    slow, fast := n, getNext(n)
    for fast != 1 && slow != fast {
        slow = getNext(slow)
        fast = getNext(getNext(fast))
    }
    return fast == 1
}

File: sol/solution/0201-0300/0202/Solution.py
class Solution:
    def isHappy(self, n: int) -> bool:
        def get_next(num):
            return sum(int(d) ** 2 for d in str(num))
        slow = n
        fast = get_next(n)
        while fast != 1 and slow != fast:
            slow = get_next(slow)
            fast = get_next(get_next(fast))
        return fast == 1
